src/handle_action.cpp

```diff
@@ -263,6 +263,10 @@ input_context game::get_player_input( std::string &action )
     const visibility_variables &cache = m.get_visibility_variables_cache();
     const level_cache &map_cache = m.get_cache_ref( u.posz() );
     const auto &visibility_cache = map_cache.visibility_cache;
+#if defined(TILES)
+    // Mark cata_tiles draw caches as dirty
+    tilecontext->set_draw_cache_dirty();
+#endif
```

Comment:
This is already called in the call to `update_visibility_cache` above.

Reply:
This is a special case for moving the camera.  While it does call `update_visibility_cache`, it doesn't move the player nor flag the visibility cache as dirty and hence the function returns before `set_draw_cache_dirty()` is called.